kedro-datasets/kedro_datasets/pandas/deltatable_dataset.py

```diff
@@ -0,0 +1,258 @@
+        if save_args:
+            self._save_args.update(save_args)
+
+        write_mode = self._save_args.get("mode", None)
+        if write_mode not in self.ACCEPTED_WRITE_MODES:
+            raise DataSetError(
+                f"Write mode {write_mode} is not supported, "
+                f"Please use any of the following accepted modes "
+                f"{self.ACCEPTED_WRITE_MODES}"
+            )
+
+        self._version = self._load_args.get("version", None)
+
+        if self._filepath and self._catalog_type:
+            raise DataSetError(
+                "DeltaTableDataSet can either load from "
+                "filepath or catalog_type. Please provide "
+                "one of either filepath or catalog_type."
+            )
+
+        if self._filepath:
+            try:
+                self._delta_table = DeltaTable(
+                    table_uri=self._filepath,
+                    storage_options=self.fs_args,
+                    version=self._version,
+                )
+            except TableNotFoundError:
+                self.is_empty_dir = True
+        else:
+            self._delta_table = DeltaTable.from_data_catalog(
+                data_catalog=DataCatalog[self._catalog_type],
+                data_catalog_id=self._catalog_name,
+                database_name=self._database,
+                table_name=self._table,
+            )
+
+    @property
+    def fs_args(self) -> Dict[str, Any]:
+        """Appends and returns filesystem credentials to fs_args."""
+        fs_args = deepcopy(self._fs_args)
+        fs_args.update(self._credentials)
+        return fs_args
+
+    @property
+    def schema(self) -> Dict[str, Any]:
+        """Returns the schema of the DeltaTableDataSet as a dictionary."""
+        return self._delta_table.schema().json()
+
+    @property
+    def metadata(self) -> Metadata:
+        """Returns the metadata of the DeltaTableDataSet as a dictionary.
+        Metadata contains the following:
+        1. A unique id
+        2. A name, if provided
+        3. A description, if provided
+        4. The list of partition_columns.
+        5. The created_time of the table
+        6. A map of table configuration. This includes fields such as delta.appendOnly,
+            which if true indicates the table is not meant to have data deleted from it.
+
+        Returns: Metadata object containing the above metadata attributes.
+        """
+        return self._delta_table.metadata()
+
+    @property
+    def history(self) -> List[Dict[str, Any]]:
+        """Returns the history of actions on DeltaTableDataSet as a list of dictionaries."""
+        return self._delta_table.history()
+
+    def get_loaded_version(self) -> int:
+        """Returns the version of the DeltaTableDataSet that is currently loaded."""
+        return self._delta_table.version()
+
+    def _load(self) -> pd.DataFrame:
+        return self._delta_table.to_pandas()
```

Comment:
Is there any merit in making this configurable so that in the future we can use one of the other public methods e.g. `to_pyarrow()` or some future addition?

Reply:
Similar issue: https://github.com/ibis-project/ibis/issues/6351

I guess, if this is under the `kedro_datasets.pandas` though, it makes sense to do `.to_pandas()` on load. But I'd love to see more future-proof of interoperability. I think this requires some broader discussion.